app/controllers/home_controller.rb

```diff
@@ -1,2 +1,33 @@
 class HomeController < ApplicationController
+  def index 
+    @products = ProductFilterService.new(params).filter
+
+    @products = filter_by_category(@products, params[:category])
+    @products = filter_by_vegetarian(@products, params[:vegetarian])
```

Comment:
You're doing the filtering already in `ProductFilterService`, do you still need this?